File: src/main/java/uk/co/alt236/gradlecache2mvn/core/artifacts/gradle/GradleMavenArtifactGroup.java
package uk.co.alt236.gradlecache2mvn.core.artifacts.gradle;

import uk.co.alt236.gradlecache2mvn.core.artifacts.ArtifactFile;
import uk.co.alt236.gradlecache2mvn.core.artifacts.MavenArtifact;

import java.util.Collection;
import java.util.Collections;

public class GradleMavenArtifactGroup implements MavenArtifact {

    private final String groupId;
    private final String artifactId;
    private final String version;
    private final Collection<ArtifactFile> files;

    public GradleMavenArtifactGroup(String groupId,
                                    String artifactId,
                                    String version,
                                    Collection<ArtifactFile> files) {
        this.groupId = groupId;
        this.artifactId = artifactId;
        this.version = version;
        this.files = Collections.unmodifiableCollection(files);
    }

    @Override
    public String getArtifactId() {
        return artifactId;
    }

    @Override
    public String getVersion() {
        return version;
    }

    @Override
    public String getGroupId() {
        return groupId;
    }

    public Collection<ArtifactFile> getFiles() {
        return files;
    }

    public String getGradleDeclaration() {
        return "'" + groupId + ":" + artifactId + ":" + version + "'";
    }

    @Override
    public String toString() {
        return "GradleMavenArtifactGroup{" +
                "groupId='" + groupId + '\'' +
                ", artifactId='" + artifactId + '\'' +
                ", version='" + version + '\'' +
                ", files=" + files +
                '}';
    }
}

File: src/main/java/uk/co/alt236/gradlecache2mvn/core/artifacts/gradle/GradleMavenArtifactGroup.kt
package uk.co.alt236.gradlecache2mvn.core.artifacts.gradle

import uk.co.alt236.gradlecache2mvn.core.artifacts.ArtifactFile
import uk.co.alt236.gradlecache2mvn.core.artifacts.MavenArtifact
import java.util.*

data class GradleMavenArtifactGroup(private val groupId: String,
                                    private val artifactId: String,
                                    private val version: String,
                                    private val files: Collection<ArtifactFile>) : MavenArtifact {

    val artifacts: MutableCollection<ArtifactFile> = Collections.unmodifiableCollection(files)

    override fun getArtifactId(): String {
        return artifactId
    }

    override fun getVersion(): String {
        return version
    }

    override fun getGroupId(): String {
        return groupId
    }

    override fun getGradleDeclaration(): String {
        return "'$groupId:$artifactId:$version'"
    }

    override fun toString(): String {
        return "GradleMavenArtifactGroup{" +
                "groupId='" + groupId + '\'' +
                ", artifactId='" + artifactId + '\'' +
                ", version='" + version + '\'' +
                ", artifacts=" + artifacts +
                '}'
    }
}